README.md

```diff
@@ -29,44 +30,93 @@ Note that `words` should be automatically installed by `pip` when you follow
 the installation instructions for this package.
 
 ## Directory Structure
+
 * `analysis_config/` contains a configuration file which can be used for analysis of cost, latency, and performance metrics for one set of hyperparameters for a particular study. This config is used in the script scripts/anaylsis_for_one_study.py
 * `config/` contains configuration files which specify hyperparamter settings for running experiments.
 * `hypothesis_config/` contains a configuration file which specifies filtering criterias for generating figures for various hypotheses.
-* `notebooks/ ` has a Jupyter notebook for generating figures that are used in the research paper
+* `notebooks/` has a Jupyter notebook for generating figures that are used in the research paper
 * `scripts/` contains Python scripts for running experiments, post-processing the results, and analysis of results
 * `setlexsem/` is the module which has all the important functions and utils for analysis, experimentation, generation of data, samplers.
-    * `analyze` contains code for error_analysis of post-processed results, visualizaiton code and utils neeeded for generating figures for hypothesis.
-    * `experiment` contains code for invoking LLMs and running experiments for a particular hypothesis/study.
-    * `generate` contains code for generating data, sample synthetic sets, prompts and utils needed for data generation.
-    * `prepare` contains helper functions for partitioning words according to their frequencies.
+  * `analyze` contains code for error_analysis of post-processed results, visualizaiton code and utils neeeded for generating figures for hypothesis.
+  * `experiment` contains code for invoking LLMs and running experiments for a particular hypothesis/study.
+  * `generate` contains code for generating data, sample synthetic sets, prompts and utils needed for data generation.
+  * `prepare` contains helper functions for partitioning words according to their frequencies.
+
+## Data Generation
+
+### Numbers and Words
+
+To generate your own data, you can run the following:
+
+```bash
+python setlexsem/generate/generate_data.py --config_path "configs/data_generation/numbers_and_words.yaml" --seed_value 292 --save_data 1
+```
+
+### Deciles
+
+To generate data based on `deciles`, you need to have `wget` installed (`brew install wget`). After installing `wget`, you need to create the `deciles.json` file. This process may take ~5-10 minutes.
+
+```bash
+scripts/raw-data-google-ngram-run.sh
+scripts/partition_words.py --args ...
+python scripts/partition_words.py \
+    --k 10 \
+    --google-ngram-path PATH/googlebooks-eng-all-1gram-20120701.filtered \
+    --output-path deciles.json
+```
+
+Then, you can run the following to generate data:
+
+```bash
+python setlexsem/generate/generate_data.py --config_path "configs/data_generation/deciles.yaml" --seed_value 292 --save_data 1
+```
+
+## Prompt Generation
```

Comment:
Should we have a sub-section for each type of lexical or semantic variation we evaluate in the paper?

Reply:
Yes, that'll be great addition. Let's have it as a to-do.